messaging/src/androidTest/java/io/lantern/messaging/MessagingTest.kt

```diff
@@ -2117,6 +2123,56 @@ class MessagingTest : BaseMessagingTest() {
         }
     }
 
+    @Test
+    fun testRecovery() {
+        testInCoroutine {
+            newDB.use { dogDB ->
+                newDB.use { cat1DB ->
+                    newDB.use { cat2DB ->
+                        newMessaging(dogDB, "dog").with { dog ->
+                            newMessaging(cat1DB, "cat").with { cat1 ->
+                                newMessaging(cat2DB, "cat").with { cat2 ->
+                                    val dogId = dog.myId.id
+                                    val cat1Id = cat1.myId.id
+
+                                    dog.addOrUpdateDirectContact(cat1Id)
+                                    val dogContact = cat1.addOrUpdateDirectContact(dogId)
+                                    sendAndVerify(
+                                        "dog sends a message to cat 1",
+                                        dog,
+                                        cat1,
+                                        "hi cat"
+                                    )
+
+                                    sendAndVerify(
+                                        "cat1 sends a message to dog",
+                                        cat1,
+                                        dog,
+                                        "hi dog"
+                                    )
+
+                                    cat1.recover(cat2.recoveryCode)
```

Comment:
I'm a bit confused: how is this working? `cat2` has no relation to `cat1`. Why is `cat2`'s recoveryKey recovering `cat1`?

Also, changing this to `cat1.recover(cat1.recoveryCode)` would yield the same result, right?